rfcs/rfc-009-staking.md

```diff
@@ -0,0 +1,59 @@
+
+- Align validator's incentives with that of the greater network through skin-in-the-game deposits at risk
+- Avoid 'nothing at stake' branch voting issues by implementing slashing rules aimed at promoting branch convergence
+- Provide an avenue for validator rewards provided as a function of validator participation in the network.
+- ?network security?
+
+While many of the details of the specific implementation are currently under consideration and are expected to come into focus through specific modeling studies and parameter exploration on the Solana testnet, we outline here our current thinking on the main components of the PoS system. Much of this thinking is based on the current status of Casper FFG, with optimizations and specific attributes to be modified as is allowed by Solana's PoH blockchain data structure.
+
+### General Overview
+Solana's validation network design is based on a rotating, stake-weight randomly selected leader broadcasting network transactions in a Proof of History data structure to validating nodes. These nodes, upon receiving the leader's broadcast, have the opportunity to vote on the current state and PoH height by signing a transaction into the PoH stream.
+
+To become a Solana validator, a network node must deposit/lock-up some amount of SOL in a contract. This SOL will not be accessible for a specific time period. The precise duration of the staking lockup period have not been determined, however we can consider three phases of this time:
+
+- *Warm-up period*: which SOL is deposited and inaccessible to the node, however PoH transaction validation has not begun. Most likely on the order of days to weeks
+- *Validation period*: a minimum duration for which the deposited SOL will be inaccessible, at risk of slashing (see slashing rules below) and earning network rewards for the validator's participation. Likely duration of months to a year.
+- *Cool-down period*: a duration of time following the submission of a 'withdrawal' transaction. During this period validation responsibilities have been removed and the funds continue to be inaccessible. Accumulated rewards should be delivered at the end of this period, along with the return of the initial deposit.
+
+Solana's trustless sense of time and ordering provided by it's PoH data structure, along with it's [avalanche](https://www.youtube.com/watch?v=qt_gDRXHrHQ&t=1s) data broadcast and transmission design, should provide sub-second finality times that scale with the log of the number of nodes in the network. This means we shouldn't have to restrict the number of validating nodes with a prohibitive 'minimum deposits' and expect nodes to be able to become validators with nominal amounts of SOL staked. This should also render validation pools unnecessary and remove the concern for needing to put slashable stake at risk while relying on others to play by the rules.
+
+### Rewards
```

Comment:
How about a title "Stake-weighted Rewards" since first sentence below describes what is meant by stake-weighting.